packages/eventual-send/src/index.js

```diff
@@ -445,24 +512,20 @@ export function makeHandledPromise() {
       }
 
       // This contestant tries to win with the target's resolution.
-      HandledPromise.resolve(p)
+      staticMethods
+        .resolve(p)
         .then(o => win('forwardingHandler', forwardingHandler, o))
         .catch(lose);
 
       // This contestant sleeps a turn, but then tries to win immediately.
-      HandledPromise.resolve()
+      staticMethods
+        .resolve()
         .then(() => {
           p = shorten(p);
-          const unsettledHandler = promiseToUnsettledHandler.get(p);
-          if (
-            unsettledHandler &&
-            typeof unsettledHandler[operation] === 'function'
-          ) {
-            // and resolve to the answer from the specific unsettled handler,
-            // opArgs are something like [prop] or [method, args],
-            // so we don't risk the user's args leaking into this expansion.
-            // eslint-disable-next-line no-use-before-define
-            win('unsettledHandler', unsettledHandler, p);
+          const pendingHandler = promiseToPendingHandler.get(p);
+          if (pendingHandler) {
+            // resolve to the answer from the specific pending handler,
+            win('pendingHandler', pendingHandler, p);
           } else if (Object(p) !== p || !('then' in p)) {
```

Comment:
Should use thenable test equivalent to `Promise.resolve`'s. Perhaps

```suggestion
          } else if (p && typeof p.then === 'function') {
```

or somehow try for equivalence with your own `isFrozenPromiseThen` perhaps?

Reply:
Suggestion taken, but inverted as that's what we need here.